### ISSUE TITLE
Allow IAM Role ARN to be passed into CloudWatch Scheduled Event creation

### ISSUE THREAD

**Issue by atheiman** [horike37/serverless-step-functions#128]
# This is a Feature Proposal

## Description

Users should be able to specify an existing IAM Role ARN in scheduled events rather than having one generated. There is currently an IAM role defined in `iamRoleTemplate` of `compileScheduledEvents.js`:
https://github.com/horike37/serverless-step-functions/blob/673f26b9c4da1ea7b81b55b43b938aa6026fd15f/lib/deploy/events/schedule/compileScheduledEvents.js#L109
This is bolted onto the CloudFormation template generated by `serverless` deploy. Our automation that deploys our Serverless projects does not have permission to deploy these IAM roles so we get this error:

```
An error occurred: myStepFuncScheduleToStepFunctionsRole - 
API: iam:CreateRole User: arn:aws:sts::<accountId>:assumed-role/<myServerlessDeploymentRole> 
is not authorized to perform: iam:CreateRole 
on resource: arn:aws:iam::<accountId>:role/<my-service-name>-myStepFuncScheduleToSt-RANDOMSTRING.
```

Our infrastructure team manages IAM resources, they are able to build us the needed role and we would then specify the Role ARN in the `serverless.yml`:

```yaml
stepFunctions:
  stateMachines:
    myStepFunc:
      role: arn:aws:iam::<accountId>:role/serverless/ScheduledEventRole
      definition:
        # ...
```

## Additional Data

`"serverless": "^1.27.3"`
`"serverless-step-functions": "^1.4.1"`

**Comment by stevecaldwell77** [comment]
I created a PR for this, let me know your thoughts.

**Comment by theburningmonk** [comment]
The change has been published as 1.17.0. This seems like a good change and we can apply this to the cloudwatch event as well. Fancy making another PR? ;-)
